book/src/anchoredkeys.md

```diff
@@ -0,0 +1,21 @@
+# Anchored keys
+Rather than dealing with just keys, we introduce the notion of an *anchored key*, which is a pair consisting of an origin specifier and a key, i.e.
+
+```
+type AnchoredKey = (Origin, Key)
+type Key = String
+```
+
+An *origin* is a triple consisting of a numeric identifier called the *origin ID*, a string called the *origin name* (omitted in the backend) and another numeric identifier called the *gadget ID*, which identifies the means by which the value corresponding to a given key is produced.
```

Comment:
Is the gadget ID included in the origin in the backend?
What is the purpose of the gadget ID in the Anchored Key?